File: CS/WinFormsDashboard_RecalculateTotals/Classes/FormatHelper.cs
using DevExpress.DashboardCommon.ViewerData;

namespace WinFormsDashboard_RecalculateTotals {
    class FormatHelper {
        public static string Format(object dataItem, object value) {
            if (dataItem is DimensionDescriptor)
                return ((DimensionDescriptor)dataItem).Format(value);
            if (dataItem is MeasureDescriptor)
                return ((MeasureDescriptor)dataItem).Format(value);
            return null;
        }
    }
}

File: VB/WinFormsDashboard_RecalculateTotals/Classes/FormatHelper.vb
Imports DevExpress.DashboardCommon.ViewerData

Namespace WinFormsDashboard_RecalculateTotals

    Friend Class FormatHelper

        Public Shared Function Format(ByVal dataItem As Object, ByVal value As Object) As String
            If TypeOf dataItem Is DimensionDescriptor Then Return CType(dataItem, DimensionDescriptor).Format(value)
            If TypeOf dataItem Is MeasureDescriptor Then Return CType(dataItem, MeasureDescriptor).Format(value)
            Return Nothing
        End Function
    End Class
End Namespace
